niri-ipc/src/lib.rs

```diff
@@ -1156,6 +1169,26 @@ pub enum Event {
         /// Id of the newly focused window, or `None` if no window is now focused.
         id: Option<u64>,
     },
+    /// Resize multiple windows at once. Some operations can cause many resizes at once, and they
+    /// are grouped together to avoid spamming events.
+    ///
+    /// Example: resizing a window in a column with many windows resizes each window.
+    BatchResizeTiles {
+        /// Pairs consisting of an id and the new size of each window.
+        id_size_pairs: Vec<(u64, (f64, f64))>,
+    },
```

Comment:
```
WindowPositionsAndSizesChanged {
    changes: Vec<(u64, PositionAndSize)>,
}
```
I think like this, and put all position and size related properties into a new struct, maybe? Also inline that struct into Window instead of individual fields. What to call it though? Geometry comes to mind, but it's also a Wayland-specific different concept so...